app/jobs/signal_adapter/receive_polling_job.rb

```diff
@@ -44,6 +44,8 @@ def perform(*_args)
     def request_new_messages
       url = URI.parse("#{Setting.signal_cli_rest_api_endpoint}/v1/receive/#{Setting.signal_server_phone_number}")
       res = Net::HTTP.get_response(url)
+      raise SignalAdapter::ServerError if (500..509).include? res.code.to_i
```

Comment:
I think this won’t work. `signal-cli-rest-api` doesn’t return a 5xx response in this case. Instead, `signal-cli-rest-api` returns an exception that indicates that the Signal service has responded with a 5xx status code.

I also think we could maybe handle this with less complexity? I’d suggest simply raising an exception in case of a non-successful response from `signal-cli-rest-api`.

```suggestion
      res.value # raises an exception if response code is not 2xx
```

… then rescuing from 4xx response to ensure the job completes successfully and no errors are reported.

```ruby
 rescue_from(Net::HTTPClientException) do |exception|
    # In case the request fails, it’s likely due to the Signal service being
    # unavailable for a short period. As the polling job is run repeatedly in
    # a short interval, we simply ignore this error. If receiving messages fails
    # persistently, we will be notified by our cron monitoring service. Note that 
    # `signal-cli-rest-api` returns a 400 error (even if a 500 error would be more
    # appropriate semantically).
  end
```

Reply:
> I think this won’t work. signal-cli-rest-api doesn’t return a 5xx response in this case. Instead, signal-cli-rest-api returns an exception that indicates that the Signal service has responded with a 5xx status code.

Yeah, I remember that from the docs, now that you mention it.

> also think we could maybe handle this with less complexity? I’d suggest simply raising an exception in case of a non-successful response from signal-cli-rest-api.

I thought about just rescuing, but I favored the "more complex", but more explicit way of handling it. Note, that we could handle this specific error by just adding a guard clause to the `SignalAdapter::Inbound#perform` method that returns if the `signal_message` does not respond_to? :with_indifferent_access. This would be less complex than both options.

But again, I think setting up a class for `IgnoredErrors` is more explicit and I can imagine we might use it in other places where we want to ignore third party servers not being available for some period of time. I also don't like calling `res.value` because we know that it might raise an exception and then needed to add a comment to explain what we are trying to do. In general, if you feel the need to write a comment to yourself or other devs, maybe there is a more explicit way to handle the same thing.

Reply:
> I thought about just rescuing, but I favored the "more complex", but more explicit way of handling it.

Fine with me if you prefer that!

> and then needing to add a comment to explain what we are trying to do. In general, if you feel the need to write a comment to yourself or other devs, maybe there is a more explicit way to handle the same thing.

I thought it might be valuable to add a comment to explain *why* we ignore the error (and why we get a 4xx error when the Signal service is unavailable), because that’s something that isn’t intuitive. The reason for that lies in a third-party component, so no matter how well our own code is structured, it will still be confusing.